src/stream.rs

```diff
@@ -15,4 +17,48 @@ impl AnyStream {
     pub fn new<S: Stream + 'static>(stream: S) -> Self {
         Self(Box::pin(stream))
     }
+
+    /// Transfers at least one byte from the stream into the read buffer.
```

Comment:
How about ...

```suggestion
    /// Reads bytes into the buffer returning [`StreamError::Closed`] when no bytes were read, e.g., due to EOF.
```

Shall we also return `Ok(byte_count)`?

Reply:
I tried another wording.